sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -77,10 +81,13 @@
      * @param messageSerializer Serializer to translate objects to and from proton-j {@link Message messages}.
      * @param product The name of the product this connection is created for.
      * @param clientVersion The version of the client library creating the connection.
+     * @param senderSettleMode to set as {@link SenderSettleMode} on sender.
+     * @param receiverSettleMode to set as {@link ReceiverSettleMode} on receiver.
      */
     public ReactorConnection(String connectionId, ConnectionOptions connectionOptions, ReactorProvider reactorProvider,
         ReactorHandlerProvider handlerProvider, TokenManagerProvider tokenManagerProvider,
-        MessageSerializer messageSerializer, String product, String clientVersion) {
+        MessageSerializer messageSerializer, String product, String clientVersion,
+        SenderSettleMode senderSettleMode, ReceiverSettleMode receiverSettleMode) {
```

Comment:
Why does a connection know about these modes? I'd expect them to be parameters when creating a receiver or sender or when creating the management node itself.